File: server/app/domain/models/video.py
from pydantic import BaseModel
from typing import Any, Optional

class Video(BaseModel):
    id: Optional[int]
    user_id: str
    title: str
    description: str
    url: str
    created_at: Optional[str] = ''
    updated_at: Optional[str] = ''


    def model_dump(self) ->  dict[str, Any]:
        return {
            'id': self.id,
            'user_id': self.user_id,
            'title': self.title,
            'description': self.description,
            'url': self.url,
            'created_at': self.created_at,
            'updated_at': self.updated_at,
        }

class UpdateVideoRequest(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    url: Optional[str] = None
    updated_at: Optional[str] = None


    def model_dump(self) ->  dict[str, Any]:
        return {
            'title': self.title,
            'description': self.description,
            'url': self.url,
            'updated_at': self.updated_at,
        }

File: app/lib/domain/entities/video.dart
import 'package:json_annotation/json_annotation.dart';

part 'video.g.dart';

@JsonSerializable()
class Video {
  final int id;
  final String userId;
  final String title;
  final String description;
  final String url;
  final String createdAt;
  final String updatedAt;

  const Video({
    required this.id,
    required this.userId,
    required this.title,
    required this.description,
    required this.url,
    this.createdAt = "",
    this.updatedAt = "",
  });

  factory Video.fromJson(Map<String, dynamic> json) => _$VideoFromJson(json);

  Map<String, dynamic> toJson() => _$VideoToJson(this);

  Video copyWith({
    int? id,
    String? userId,
    String? title,
    String? description,
    String? url,
    String? createdAt,
    String? updatedAt,
  }) {
    return Video(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      title: title ?? this.title,
      description: description ?? this.description,
      url: url ?? this.url,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  String toString() {
    return 'Video(id: $id, userId: $userId, title: $title, description: $description, url: $url, createdAt: $createdAt, updatedAt: $updatedAt)';
  }

  @override
  bool operator ==(covariant Video other) {
    if (identical(this, other)) return true;

    return other.id == id &&
        other.userId == userId &&
        other.title == title &&
        other.description == description &&
        other.url == url &&
        other.createdAt == createdAt &&
        other.updatedAt == updatedAt;
  }

  @override
  int get hashCode {
    return id.hashCode ^
        userId.hashCode ^
        title.hashCode ^
        description.hashCode ^
        url.hashCode ^
        createdAt.hashCode ^
        updatedAt.hashCode;
  }
}
